python/nano/src/bigdl/nano/utils/pytorch/model_info.py

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import inspect
+from typing import Optional, Union
+
+import torch
+from bigdl.nano.pytorch.model import AcceleratedLightningModule
+
+
+class ModelInfo:
+    def __init__(self, model):
+        self.forward_fullargspec = ModelInfo.get_forward_fullargspec(model)
+
+        """
+        This function is to get all the arguments(excepts *args and **kwargs)
+        It will return a list of arg name
+        E.g.
+        def forward(self, a, b=1, c: int = 3, *args, **kwargs):
+           pass
+        it will return ['a', 'b', 'c']
+        """
+        self.forward_args = self.forward_fullargspec.args[1:]
+
+        """
+        This function is to get all the defaults
+        It will return a list of default values
+        E.g.
+        def forward(self, a, b=1, c: int = 3, *args, **kwargs):
+            pass
+        it will return (1, 3)
+        """
+        self.forward_defaults = self.forward_fullargspec.defaults
+
+        """
+        This function is to get all the annotations
+        It will return a dict of {args: annotations}
+        E.g.
+        def forward(self, a, b=1, c: int = 3, *args, **kwargs):
+            pass
+        it will return {'c': <class 'int'>}
+        """
+        self.forward_annotations = self.forward_fullargspec.annotations
+
+    @staticmethod
+    def get_forward_fullargspec(model):
```

Comment:
Besides, there are already `get_forward_args` and  `get_condition_args` in <https://github.com/intel-analytics/BigDL/blob/5b99f80837186865780b18818545afb0c2ff49d1/python/nano/src/bigdl/nano/utils/pytorch/inspect.py#L22>, why not use them directly ?

Reply:
Exactly, what we want to do is to aggregate data preprocess logic in one or two classes. In addition, the previous `get_forward_args`, `get_forward_defaults` and `get_forward_annotations` have redundancy code.

Reply:
oh, I see